src/rpcclient.cpp

```diff
@@ -134,9 +136,20 @@ Array RPCConvertValues(const std::string &strMethod, const std::vector<std::stri
 
         // parse string as JSON, insert bool/number/object/etc. value
         else {
+            //according to rfc4627 null, true, false are not valid json strings
             Value jVal;
-            if (!read_string(strVal, jVal))
-                throw runtime_error(string("Error parsing JSON:")+strVal);
+            if(strVal == "null")
+                jVal.setNull();
+            else if(strVal == "true")
+                jVal.setBool(true);
+            else if(strVal == "false")
+                jVal.setBool(false);
+            else
+            {
+                if (!jVal.read(strVal) || (jVal.isNull() && strVal.size() > 0))
+                    if(!jVal.setNumStr(strVal) || jVal.isNull())
+                        throw runtime_error(string("Error parsing JSON:")+strVal);
+            }
```

Comment:
I had to extend this to gain compatibility between JSON Spirit and UniValue. UniValue is more strict in case of JSON encoding.